subsys/net/ip/tcp.c

```diff
@@ -229,10 +229,17 @@ static void tcp_retry_expired(struct k_work *work)
 		pkt = CONTAINER_OF(sys_slist_peek_head(&tcp->sent_list),
 				   struct net_pkt, sent_list);
 
-		if (net_pkt_sent(pkt)) {
-			do_ref_if_needed(tcp, pkt);
-			net_pkt_set_sent(pkt, false);
-		}
+		/* In the retry case, the original ref (when the packet
+		 * was created) is set to 1. That original ref was
+		 * subtracted when the packet was sent by the driver.
```

Comment:
I'm not sure description here is clear. "That original ref was subtracted" - subtracted from what? Maybe it was decremented after all?

Reply:
Yes, decremented is better, will fix.